app/src/main/java/chat/rocket/android/server/infraestructure/SharedPreferencesAccountsRepository.kt

```diff
@@ -6,41 +6,37 @@ import chat.rocket.android.server.domain.AccountsRepository
 import chat.rocket.android.server.domain.model.Account
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.Types
-import kotlinx.coroutines.experimental.CommonPool
-import kotlinx.coroutines.experimental.withContext
+
+private const val ACCOUNTS_KEY = "ACCOUNTS_KEY"
 
 class SharedPreferencesAccountsRepository(
     private val preferences: SharedPreferences,
     private val moshi: Moshi
 ) : AccountsRepository {
 
-    override suspend fun save(newAccount: Account) {
-        withContext(CommonPool) {
-            val accounts = load()
+    override fun save(newAccount: Account) {
+        val accounts = load()
 
-            val newList = accounts.filter { account -> newAccount.serverUrl != account.serverUrl }
-                    .toMutableList()
-            newList.add(0, newAccount)
-            save(newList)
-        }
+        val newList = accounts.filter { account -> newAccount.serverUrl != account.serverUrl }
+            .toMutableList()
+        newList.add(0, newAccount)
+        save(newList)
     }
 
-    override suspend fun load(): List<Account> = withContext(CommonPool) {
+    override fun load(): List<Account> {
```

Comment:
Shouldn't this be async/coroutine, since we are reading the SharedPrefs and doing some JSON mapping here?